src/QsCompiler/Transformations/ClassicallyControlled.cs

```diff
@@ -413,41 +420,92 @@ private QsStatement ConvertConditionalToControlCall(QsStatement statement)
                 }
 
                 /// <summary>
-                /// Checks if the expression is an equality comparison where one side is a Result literal.
-                /// If it is, returns true along with the Result literal and the other expression in the
-                /// equality, otherwise returns false with nulls.
+                /// Checks if the expression is an equality or inequality comparison where one side is a
+                /// Result literal. If it is, returns true along with the Result literal and the other
+                /// expression in the (in)equality, otherwise returns false with nulls. If it is an
+                /// inequality, the returned result value will be the opposite of the result literal found.
                 /// </summary>
-                private (bool, QsResult, TypedExpression) IsConditionedOnResultLiteralExpression(TypedExpression expression)
+                private bool IsConditionedOnResultLiteralExpression(TypedExpression expression, out QsResult literal, out TypedExpression nonLiteral)
```

Comment:
What happens if both expressions are literals? The name `nonLiteral` wouldn't be correct in that case.